chunkset.c

```diff
@@ -5,7 +5,7 @@
 #include "zbuild.h"
 #include "zutil.h"
 
-#ifdef UNALIGNED_OK
+#ifdef UNALIGNED64_OK
 typedef uint64_t chunk_t;
```

Comment:
We can do:
```
#if defined(UNALIGNED64_OK)
typedef uint64_t  chunk_t;
#elif defined(UNALIGNED_OK)
typedef uint32_t chunk_t;
#else
typedef uint8_t chunk_t;
#endif
```

But the functions below would also need to support `chunk_t` being `uint32_t`.

Reply:
True, but I'd have to actually grok what's happening in there ^^.

I'll take a closer look at it after dinner, but no promises ;).

Reply:
Does that look okay? Not *quite* confident about `chunkmemset_4`, but the rest is trivial enough (maybe :D).

```diff
diff --git a/chunkset.c b/chunkset.c
index 121f3a4..423c69e 100644
--- a/chunkset.c
+++ b/chunkset.c
@@ -5,8 +5,10 @@
 #include "zbuild.h"
 #include "zutil.h"
 
-#ifdef UNALIGNED64_OK
+#if defined(UNALIGNED64_OK)
 typedef uint64_t chunk_t;
+#elif defined(UNALIGNED_OK)
+typedef uint32_t chunk_t;
 #else
 typedef uint8_t chunk_t[8];
 #endif
@@ -16,18 +18,24 @@ typedef uint8_t chunk_t[8];
 #define HAVE_CHUNKMEMSET_8
 
 static inline void chunkmemset_1(uint8_t *from, chunk_t *chunk) {
-#ifdef UNALIGNED64_OK
+#if defined(UNALIGNED64_OK)
     *chunk = 0x0101010101010101 * (uint8_t)*from;
+#elif defined(UNALIGNED_OK)
+    *chunk = 0x01010101 * (uint8_t)*from;
 #else
     memset(chunk, *from, sizeof(chunk_t));
 #endif
 }
 
 static inline void chunkmemset_4(uint8_t *from, chunk_t *chunk) {
-#ifdef UNALIGNED64_OK
+#if defined(UNALIGNED64_OK)
     uint32_t half_chunk;
     half_chunk = *(uint32_t *)from;
     *chunk = 0x0000000100000001 * (uint64_t)half_chunk;
+#elif defined(UNALIGNED_OK)
+    uint16_t half_chunk;
+    half_chunk = *(uint16_t *)from;
+    *chunk = 0x00010001 * (uint32_t)half_chunk;
 #else
     uint8_t *chunkptr = (uint8_t *)chunk;
     memcpy(chunkptr, from, 4);
@@ -36,8 +44,10 @@ static inline void chunkmemset_4(uint8_t *from, chunk_t *chunk) {
 }
 
 static inline void chunkmemset_8(uint8_t *from, chunk_t *chunk) {
-#ifdef UNALIGNED64_OK
+#if defined(UNALIGNED64_OK)
     *chunk = *(uint64_t *)from;
+#elif defined(UNALIGNED_OK)
+    *chunk = *(uint32_t *)from;
 #else
     memcpy(chunk, from, sizeof(chunk_t));
 #endif
@@ -48,8 +58,10 @@ static inline void loadchunk(uint8_t const *s, chunk_t *chunk) {
 }
 
 static inline void storechunk(uint8_t *out, chunk_t *chunk) {
-#ifdef UNALIGNED64_OK
+#if defined(UNALIGNED64_OK)
     *(uint64_t *)out = *chunk;
+#elif defined(UNALIGNED_OK)
+    *(uint32_t *)out = *chunk;
 #else
     /* Cast to chunk_t pointer to avoid compiler error on MSVC ARM */
     memcpy((chunk_t *)out, chunk, sizeof(chunk_t ));
```

A very quick bsdtar test on a random tarball appears successful with this, FWIW ;).

Reply:
In `chunkmemset_4`:
```
+#elif defined(UNALIGNED_OK)
+    uint16_t half_chunk;
+    half_chunk = *(uint16_t *)from;
+    *chunk = 0x00010001 * (uint32_t)half_chunk;
```
Should just be:
```
+#elif defined(UNALIGNED_OK)
+    *chunk = *(uint32_t *)from;
```
And for `chunkmemset_8`, I don't think it should exist if defined `UNALIGNED64_OK` is not defined and neither should be `HAVE_CHUNKMEMSET_8`. So wrap those in `#ifdef UNALIGNED64_OK`..

Reply:
Should be taken care of w/ e09ef09, thanks for the pointers (ha!)!